src/api/api.cc

```diff
@@ -12104,6 +12110,41 @@ extern "C" DLLEXPORT void V8RecordReplayExitReplayCode() {
 }
 
 
+extern "C" DLLEXPORT void V8RecordReplayBeginAssertBufferAllocations(const char* issueLabel) {
+  if (!recordreplay::IsRecordingOrReplaying() || recordreplay::AreAssertsDisabled()) {
+    return;
+  }
+  size_t enabled = recordreplay::AutoAssertBufferAllocations::GetEnabled();
+  
+  if (!enabled) {
+    base::Thread::SetThreadLocal(gBufferAllocationsLabelLSKey, 
+      reinterpret_cast<void*>(const_cast<char*>(issueLabel))
+    );
+  }
+  ++enabled;
+  base::Thread::SetThreadLocal(gBufferAllocationsEnabledLSKey, 
+    reinterpret_cast<void*>(enabled)
+  );
+}
+
+extern "C" DLLEXPORT void V8RecordReplayEndAssertBufferAllocations() {
+  if (!recordreplay::IsRecordingOrReplaying() || recordreplay::AreAssertsDisabled()) {
+    return;
+  }
+  
+  size_t enabled = recordreplay::AutoAssertBufferAllocations::GetEnabled();
+  --enabled;
+  if (!enabled) {
+    const char* issueLabel = recordreplay::AutoAssertBufferAllocations::GetIssueLabel();
+    base::Thread::SetThreadLocal(gBufferAllocationsLabelLSKey, nullptr);
+    delete[] issueLabel;
+  }
+  base::Thread::SetThreadLocal(gBufferAllocationsEnabledLSKey, 
+    reinterpret_cast<void*>(enabled)
+  );
```

Comment:
These functions are pretty ugly.  It would be much cleaner if there was a single tls key pointing to a structure containing the assertion state.